osm2streets/src/transform/collapse_intersections.rs

```diff
@@ -157,11 +158,11 @@ pub fn collapse_intersection(streets: &mut StreetNetwork, i: NodeID) {
     }
 
     info!("Collapsing degenerate {}", i);
-    streets.intersections.remove(&i).unwrap();
     // We could be more careful merging percent_incline and osm_tags, but in practice, it doesn't
     // matter for the short segments we're merging.
-    let mut new_road = streets.roads.remove(&r1).unwrap();
-    let mut road2 = streets.roads.remove(&r2).unwrap();
+    let mut new_road = streets.remove_road(&r1);
+    let mut road2 = streets.remove_road(&r2);
+    streets.intersections.remove(&i).unwrap();
```

Comment:
Order matters! `remove_road` cleans up the two intersections, so they need to both exist until the end